lib/filterx/filterx-object.h

```diff
@@ -480,4 +480,12 @@ filterx_object_set_dirty(FilterXObject *self, gboolean value)
 
 #include "filterx-ref.h"
 
+static inline gboolean
+filterx_object_is_type_or_ref(FilterXObject *object, FilterXType *type)
```

Comment:
I wanted to avoid introducing such a function because `FilterXRef` should be completely transparent for the developer unless they want to do downcasts, but it those cases, the developer should get familiar with refs and do an explicitly unwrap on them.

I see this comes in handy in the dict implementation, but I'm wondering: can we move this into the `dict.h` file and specialize it like `is_dict_or_ref()` to make it clear that this should not be used, only in special cases?

Reply:
this is 99% used testing code, in cases where we only check if we got the right type but do not actually work with the objects themselves.

In those cases having to typecheck against both ref and the native type is pretty cumbersome.